BookReader/plugins/plugin.vendor-fullscreen.js

```diff
@@ -0,0 +1,189 @@
+    this.updateBrClasses();
+
+    this.resize();
+    this.jumpToIndex(this.currentIndex());
+
+    this.refs.$brContainer.animate({ opacity: 1 }, 400, 'linear');
+
+    this._fullWindowExitHandler = function (e) {
+      if (e.keyCode === 27) this.exitFullScreen();
+    }.bind(this);
+    $(document).on('keyup.' + event_namespace, this._fullWindowExitHandler);
+  };
+
+  BR.prototype.exitFullWindow = function() {
+    this.refs.$brContainer.css('opacity', 0);
+
+    $(document).off('keyup' + event_namespace);
+
+    this.isFullscreenActive = false;
+    this.updateBrClasses()
+
+    this.resize();
+    this.refs.$brContainer.animate({ opacity: 1 }, 400, 'linear');
+  };
+
+  BR.prototype.isFullscreen = function() {
+    return BR.util.isFullscreenActive() || this.isVendorFullscreenActive;
+  };
+
+  BR.prototype.toggleFullscreen = function() {
+    if (this.isFullscreen()) {
+      if (BR.util.fullscreenAllowed()) {
+        BR.util.exitFullscreen();
+      } else {
+        this.exitFullWindow();
+      }
+    } else {
+      if (BR.util.fullscreenAllowed()) {
+        BR.util.requestFullscreen(this.refs.$br[0]);
+      } else {
+        this.enterFullWindow();
+      }
+    }
+  };
+
+  /**
+   * Returns the DOM element being used for fullscreen.
+   * @see https://developer.mozilla.org/en-US/docs/Web/API/DocumentOrShadowRoot/fullscreenElement
+   */
+  BR.util.getFullscreenElement = function() {
+    return document.fullscreenElement ||
+           document.webkitFullscreenElement ||
+           document.mozFullScreenElement ||
+           document.msFullscreenElement;
+  };
+
+  /**
+   * Returns true if the document is in fullscreen mode.
+   */
+  BR.util.isFullscreenActive = function() {
+    var fullscreenElement = this.getFullscreenElement();
+    return fullscreenElement !== null && fullscreenElement !== undefined;
+  };
+
+  /**
+   * Exits fullscreen mode.
+   * @see https://developer.mozilla.org/en-US/docs/Web/API/Document/exitFullscreen
+   */
+  BR.util.exitFullscreen = function() {
+      if (document.exitFullscreen) {
+          document.exitFullscreen();
+      } else if (document.webkitExitFullscreen) {
+          document.webkitExitFullscreen();
+      } else if (document.mozExitFullScreen) {
+          document.mozExitFullScreen();
+      } else if (document.msExitFullscreen) {
+          document.msExitFullscreen();
+      }
+  };
+
+  /**
+   * Requests fullscreen mode for the given element
+   * @see https://developer.mozilla.org/en-US/docs/Web/API/Element/requestFullscreen
+   */
+  BR.util.requestFullscreen = function(element) {
+      if (element.requestFullscreen) {
+        element.requestFullscreen();
+      } else if (element.webkitRequestFullscreen) {
+        element.webkitRequestFullscreen();
+      } else if (element.mozRequestFullScreen) {
+        element.mozRequestFullScreen();
+      } else if (element.msRequestFullscreen) {
+        element.msRequestFullscreen();
+      }
```

Comment:
There's some spacing inconsistencies.. like 4/2 spaces here and up above.

Reply:
Yeah, there's likely some cleanup that can be done throughout. Things that stuck out while I was scanning it were addressed, but there's likely more to do with it.